src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -1264,131 +1259,55 @@ public override BoundNode VisitIndexerAccess(BoundIndexerAccess node)
             return null;
         }
 
-        protected virtual void VisitReceiverOfEventAssignmentAsRvalue(BoundEventAssignmentOperator node)
+        private void VisitArguments(ImmutableArray<BoundExpression> arguments, ImmutableArray<RefKind> refKindsOpt, MethodSymbol method)
         {
-            VisitRvalue(node.ReceiverOpt);
-        }
+            Debug.Assert(!(this is NullableWalker));  // Should call NullableWalker.VisitArguments instead, for null warnings.
 
-        private void VisitArguments(ImmutableArray<BoundExpression> arguments, ImmutableArray<RefKind> refKindsOpt, MethodSymbol method, ImmutableArray<int> argsToParamsOpt, bool expanded)
-        {
-            if (refKindsOpt.IsDefault)
-            {
-                for (int i = 0; i < arguments.Length; i++)
-                {
-                    VisitArgumentAsRvalue(arguments, i, method, argsToParamsOpt, expanded);
-                }
-            }
-            else
+            // first value and ref parameters are read...
+            for (int i = 0; i < arguments.Length; i++)
             {
-                // first value and ref parameters are read...
-                for (int i = 0; i < arguments.Length; i++)
+                RefKind refKind = GetRefKind(refKindsOpt, i);
+                if (refKind != RefKind.Out)
                 {
-                    RefKind refKind = refKindsOpt.Length <= i ? RefKind.None : refKindsOpt[i];
-                    if (refKind != RefKind.Out)
-                    {
-                        VisitArgumentAsRvalue(arguments, i, method, argsToParamsOpt, expanded);
-                    }
-                    else
-                    {
-                        VisitLvalue(arguments[i]);
-                    }
+                    VisitRvalue(arguments[i]);
                 }
-                // and then ref and out parameters are written...
-                for (int i = 0; i < arguments.Length; i++)
+                else
                 {
-                    RefKind refKind = refKindsOpt.Length <= i ? RefKind.None : refKindsOpt[i];
-
-                    // passing as a byref argument is also a potential write
-                    if (refKind != RefKind.None)
-                    {
-                        WriteArgument(arguments, i, refKind, method, argsToParamsOpt, expanded);
-                    }
+                    VisitLvalue(arguments[i]);
                 }
             }
-        }
-
-        private void VisitArgumentAsRvalue(ImmutableArray<BoundExpression> arguments, int i, MethodSymbol method, ImmutableArray<int> argsToParamsOpt, bool expanded)
-        {
-            ParameterSymbol parameter = GetCorrespondingParameter(i, method, argsToParamsOpt, ref expanded);
-            VisitArgumentAsRvalue(arguments[i], parameter, expanded);
-        }
-
-        private static ParameterSymbol GetCorrespondingParameter(int argumentOrdinal, MethodSymbol method, ImmutableArray<int> argsToParamsOpt, ref bool expanded)
```

Comment:
>o [](start = 43, length = 1)

I didn't understand why the mapping of argument to parameter is no longer needed.

Reply:
The mapping is only needed by `NullableWalker`, to report warnings based on the actual parameters, so `GetCorrespondingParameter` has been moved to `NullableWalker`.